sympy/polys/polyroots.py

```diff
@@ -902,7 +898,12 @@ def roots(f, *gens, **flags):
         if f.is_multivariate:
             raise PolynomialError('multivariate polynomials are not supported')
 
-    def _update_dict(result, currentroot, k):
+    def _update_dict(result, zeros, currentroot, k):
+        if currentroot == S.Zero:
+            if S.Zero in zeros:
+                zeros[S.Zero] += k
+            else:
+                zeros[S.Zero] = k
```

Comment:
Apparently this line isn't tested. It's not obvious to me how to hit it though...

Reply:
I don't think `zeros` should be modified here. All zero roots are extracted before anything else is done. New zero roots of modified polynomials may appear after translation/rescaling but those should not be added here as they will be non-zero after the transformations have been reverted.

Reply:
I do not recall exactly, but based on the comment https://github.com/sympy/sympy/pull/17482#issuecomment-523921526 I believe what happens is that one get pretty large expressions that SymPy cannot simplify to zero at the start. However, after some iterations it is realized that they actually are zero. So "All zero roots are extracted before anything else is done." is not how it works in practice as SymPy doesn't realize that they are zeros. Which is the root(!) cause of problem.

For the testing of that line, no idea. I either thought that there may be cases where `zeros[S.Zero]` is not initialized or found a case where it isn't while playing around.